eth/catalyst/api.go

```diff
@@ -144,7 +144,7 @@ func (api *ConsensusAPI) ForkchoiceUpdatedV1(update beacon.ForkchoiceStateV1, pa
 		}
 	}
 
-	if rawdb.ReadCanonicalHash(api.eth.ChainDb(), block.NumberU64()) != update.HeadBlockHash {
+	if rawdb.ReadCanonicalHash(api.eth.ChainDb(), block.NumberU64()) != update.HeadBlockHash || block.NumberU64() > api.eth.BlockChain().CurrentBlock().NumberU64() {
```

Comment:
The scenario is:  the local canonical hash is matched with the block, but local chain head is lower than the block.

I am wondering why it can happen in the first place. Whenever reorg happens, all the canonical hash above the new head will be deleted.